spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/TaskExecutionService.java

```diff
@@ -94,4 +94,14 @@ public interface TaskExecutionService {
 	 * @since 2.8
 	 */
 	Integer getAllTaskExecutionsCount(boolean onlyCompleted, String taskName);
+
+	/**
+	 * Returns the count of all the task execution IDs with the option to include only the completed task executions.
+	 * @param onlyCompleted filter by completed task executions
+	 * @param taskName the task name, if null then retrieve all the tasks
+	 * @param days the days before the task should ended
```

Comment:
I think "days" is a bit ambiguous - especially since this method does not include the "BeforeEndDate" suffix like the other newly introduced methods in the DAO layer. 

Wdyt about adding renaming "days" to something like ...
```java
@param includeTasksEndedMinDaysAgo only include tasks that have ended at least this many days ago
```